src/pages/Script/ScriptsComp.tsx

```diff
@@ -54,8 +64,16 @@ export const ScriptTransactions = ({ page, size }: { page: number; size: number
   const { total, ckbTransactions } = data
   const totalPages = Math.ceil(total / size)
 
+  const isChecked = restrict === true
   const onChange = (page: number) => {
-    history.push(`/${language}/script/${codeHash}/${hashType}?page=${page}&size=${size}`)
+    history.push(
+      `/${language}/script/${codeHash}/${hashType}?page=${page}&size=${size}&restrict=${isChecked.toString()}`,
+    )
+  }
```

Comment:
We can keep other query parameters by

```ts
const search = new URLSearchParams(window.location.search)
search.set('page', page) // we can also use { ...search, page, search }
search.set('size', size)
const url `/${history}/script/${codeHash}/${hashType}?{search}`
```

only page and size in the existing parameters will be updated.

Reply:
Any update on this comment @zmcNotafraid